File: Crio/crio_programming_interview_problems-master/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal.py
from crio.io import PrintMatrix
from crio.ds.Tree.TreeNode import TreeNode, readTreeReturnRoot
from Solution import inorderTraversal 

# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS


def main():
    tests = int(input())
    for _ in range(tests):
        root  = readTreeReturnRoot()
        result = inorderTraversal(root)
        PrintMatrix.OneDMatrix(result, " ")
        print()

if __name__ == '__main__':
    import sys
    sys.setrecursionlimit(30 * 1000)
    main()

File: Crio/StackAndQueue/3_BinaryTreeInorderTraversal/BinaryTreeInorderTraversal.cpp
#include <bits/stdc++.h>
#include "../crio/cpp/ds/TreeNode/TreeNode.hpp"
#include "Solution.cpp"
using namespace std;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

int main()
{
    int tests;
    cin >> tests;
    while (tests--)
    {
        TreeNode* root  = TreeNode().readTreeReturnRoot();
        vector<int> result = BinaryTreeInorderTraversal().inorderTraversal(root);
        for ( auto it : result) {
            cout << it << " ";
        }
        cout << endl;
    }
    return 0;
}
